db_checker.py

```diff
@@ -114,24 +116,20 @@ def allowed_unit(self, unit):
         """
         This method checks that the given unit conforms to standard
         """
-        if unit in allowed_units:
-            return True
+        # expand macro $(A) to a valid unit, expand $(A=B) to B
+        u = re.sub(r'\$[({].*?=(.*)?[})]', r'\1', unit)
+        u = re.sub(r'\$[({].*?[})]', 'm', u)
 
-        # otherwise check for macro
-        if "$" in unit:
-            return True
+        # split unit amalgamations
+        units = re.split(r'[/ ()]', u)
 
-        # otherwise split unit amalgamations
-        units = re.split(r'[/ ()]', unit)
         for u in units:
+            # remove any powers of units e.g. mm^2 -> mm
+            u = re.sub(r'\^[-]?\d', '', u)
             if not (u in allowed_units):
-                # may be to the power of
-                if not (re.match(r"\^[-]?\d", u)):
-                    # may be prefixed
-                    if not (re.match(r'T|G|M|k|m|u|n|p|f', u[0]) and u[1:] in allowed_units):
-                        # special case of cm
-                        if not u == "cm":
-                            return False
+                # may be prefixed
+                if not (re.match(allowed_units_prefixes, u[0]) and u[1:] in allowed_units):
```

Comment:
A good point, given I refactored the string into a constant I should have gone further and done that